src/components/map/map-triangulated-strain-layer.tsx

```diff
@@ -0,0 +1,265 @@
+                                const responseText = element.data;
+                                const responseData = responseText.split(/\r|\n|\r/);
+                                const splitResponseData = responseData[8].split(",");
+                                const station: StationData = {
+                                    id: splitResponseData[0],
+                                    name: splitResponseData[1],
+                                    longitude: parseFloat(splitResponseData[3]) - 360,
+                                    latitude: parseFloat(splitResponseData[2]),
+                                    eastVelocity: parseFloat(splitResponseData[6]),
+                                    eastVelocityUncertainty: 0.01,
+                                    northVelocity: parseFloat(splitResponseData[5]),
+                                    northVelocityUncertainty: 0.01
+                                };
+                                outputData.push(station);
+                            }
+                        }
+                    });
+                    this.setState({data: outputData});
+                    this.buildMesh(outputData);
+                }).catch(err => {
+                    if (err.response) {
+                        console.log(err.response);
+                    } else {
+                        console.log(err);
+                    }
+                });
+
+            }
+        ).catch(err => {
+            console.log(err);
+        });
+    }
+
+    // This method creates and displays a mesh based on the GPS stations acquired from getUNAVCOData()
+    // It uses Delaunator to calculate the mesh and colors each triangle based on calculated strain
+    // The strain calculation can be found in "../../strain.ts"
+    private buildMesh(data: StationData[]) {
+        const { map } = this.props;
+
+        // Proximity based point removal
+        // GPS points that are very close to each other will produce extremely high strain values
+        // By removing these points, it becomes easier to plot the data using an infinite scale
+        // Other methods of solving this problem would be by plotting the data in a bucketed gradient
+        // e.g. 0 - 5: Blue, 5 - 50: Green, 50 - 250: Yellow, 250+: Red
+        const removablePoints: Set<string> = new Set<string>();
+        for (let i = 0; i < data.length; i++) {
+            for (let k = 0; k < data.length; k++) {
+                if (i !== k && !removablePoints.has(data[i].id) && !removablePoints.has(data[k].id)) {
+                    const dist = Math.sqrt(Math.pow(data[i].latitude - data[k].latitude, 2) +
+                                           Math.pow(data[i].longitude - data[k].longitude, 2));
+                    if (dist < 0.1) {
+                        removablePoints.add(data[i].id);
+                    }
+                }
+            }
+        }
+
+        // Output station id for all points removed from the mesh
```

Comment:
Looks like some dead code here

Reply:
I think you're right, but I'm pretty much leaving Saul's strain-layer-component as-is until we actually start working on it. I didn't have the cognitive space to try and understand it all yet.